src/modules/campaign-monitor/services/campaign-monitor.service.ts

```diff
@@ -33,12 +55,41 @@ export class CampaignMonitorService {
 		this.isEnabled = enabled;
 	}
 
-	// TODO call on status change
-	public async send(emailInfo: EmailInfo): Promise<boolean> {
+	public buildUrlToAdminVisit(): string {
+		const url = new URL(this.clientHost);
+		url.pathname = 'beheer/aanvragen';
+		return url.href;
+	}
+
+	public convertVisitToEmailTemplateData(visit: Visit): CampaignMonitorVisitData {
+		return {
+			client_firstname: visit.visitorFirstName,
+			client_lastname: visit.visitorLastName,
+			client_email: visit.visitorMail,
+			contentpartner_name: visit.spaceName,
+			contentpartner_email: visit.spaceMail,
+			request_reason: visit.reason,
+			request_time: visit.timeframe,
+			request_url: this.buildUrlToAdminVisit(), // TODO deeplink to visit & extract to shared url builder?
+			request_remark: get(visit.note, 'note', ''),
+			start_date: visit.startAt ? formatAsBelgianDate(visit.startAt, 'd MMMM yyyy') : '',
+			start_time: visit.startAt ? formatAsBelgianDate(visit.startAt, 'HH:mm') : '',
+			end_date: visit.endAt ? formatAsBelgianDate(visit.endAt, 'd MMMM yyyy') : '',
+			end_time: visit.endAt ? formatAsBelgianDate(visit.endAt, 'HH:mm') : '',
+		};
+	}
+
+	public async send(emailInfo: VisitEmailInfo): Promise<boolean> {
+		if (!emailInfo.to) {
+			// Throw exception will break too much
+			this.logger.error('Empty email address - Mail not sent', emailInfo);
```

Comment:
this log will be useless since we won't know for which user this occured. Maybe we can include the user id in this object? or will that cause issues with CM?

Reply:
updated